src/H5Epublic.h

```diff
@@ -59,8 +59,15 @@ typedef struct H5E_error2_t {
 #endif /* H5private_H */
 
 /* HDF5 error class */
+/* Extern "C" block needed to compile C++ filter plugins with some compilers */
+#ifdef __cplusplus
+extern "C" {
+#endif
 #define H5E_ERR_CLS (H5OPEN H5E_ERR_CLS_g)
 H5_DLLVAR hid_t H5E_ERR_CLS_g;
+#ifdef __cplusplus
+}
+#endif
```

Comment:
With so many macros depending on each other, I made the safest, most conservative, change.

If I move that piece of code:

```
#define H5E_ERR_CLS (H5OPEN H5E_ERR_CLS_g)
H5_DLLVAR hid_t H5E_ERR_CLS_g;
``` 

to the extern C block later on in the file, other plugins fail at compilation time.

The alternative is to extend the extern C block instead of moving the code.

That also works, but would it be acceptable?

Reply:
Let's leave it as-is and merge it.